### PR TITLE
dev/ci/integration: test against docker-compose

### PR DESCRIPTION
**Pull request by bobheadxi**
Closes https://github.com/sourcegraph/sourcegraph/issues/29413

### CODE REVIEW

**Comment by bobheadxi** [comment]
Update on this:

- it seems `docker-compose` requires baremetal to spin up an instance in a stable manner, even with reduced resource requirements and aggressive pruning
   - `GITHUB_TOKEN` is missing from the baremetal agents that is blocking backend integration and code intel from running - I'm not sure if adding this will affect the other integration tests, do you know @jhchabran ? I recall some unintentional tokens causing misbehaviour in tests before
- overall build time improvements from not waiting for `sourcegraph/server` are largely negated by:
   - `docker-compose` taking ~2.5m to spin up, as opposed to ~1.25m for server. I think there is some tweaking we might be able to do here, but I think a lot of this comes from overall higher bandwidth usage (separate images to download instead of one big one)
   - the `frontend` image taking ~9m, as opposed to ~11m for server
 - using `docker-compose` actually causes some tests to fail: https://buildkite.com/sourcegraph/sourcegraph/builds/125528#25e8a1d2-9d8b-4da1-b8c7-39917a3a6c08/758-995 . Requires further debugging

tl;dr not as happy of an outcome as expected 😢
